script/utils/GenerateConstantsJson.ts

```diff
@@ -0,0 +1,255 @@
+      result[chainName] = address;
+    }
+  }
+
+  return result;
+}
+
+// Helper function to convert L1_ADDRESS_MAP to the expected format
+function convertL1Addresses(): { [key: string]: { [key: string]: string } } {
+  const result: { [key: string]: { [key: string]: string } } = {};
+
+  for (const [chainId, addresses] of Object.entries(L1_ADDRESS_MAP)) {
+    const chainName = Object.keys(CHAIN_IDs).find(
+      (key) => CHAIN_IDs[key as keyof typeof CHAIN_IDs] === Number(chainId)
+    );
+    if (chainName) {
+      result[chainName] = addresses;
+    }
+  }
+
+  return result;
+}
+
+// Helper function to convert L2_ADDRESS_MAP to the expected format
+function convertL2Addresses(): { [key: string]: { [key: string]: string } } {
+  const result: { [key: string]: { [key: string]: string } } = {};
+
+  for (const [chainId, addresses] of Object.entries(L2_ADDRESS_MAP)) {
+    const chainName = Object.keys(CHAIN_IDs).find(
+      (key) => CHAIN_IDs[key as keyof typeof CHAIN_IDs] === Number(chainId)
+    );
+    if (chainName) {
+      result[chainName] = addresses;
+    }
+  }
+
+  return result;
+}
+
+// Helper function to convert OP_STACK_ADDRESS_MAP to the expected format
+function convertOpStackAddresses(): { [key: string]: { [key: string]: { [key: string]: string } } } {
+  const result: { [key: string]: { [key: string]: { [key: string]: string } } } = {};
+
+  for (const [hubChainId, spokeChains] of Object.entries(OP_STACK_ADDRESS_MAP)) {
+    const hubChainName = Object.keys(CHAIN_IDs).find(
+      (key) => CHAIN_IDs[key as keyof typeof CHAIN_IDs] === Number(hubChainId)
+    );
+    if (hubChainName) {
+      result[hubChainName] = {};
+      for (const [spokeChainId, addresses] of Object.entries(spokeChains)) {
+        const spokeChainName = Object.keys(CHAIN_IDs).find(
+          (key) => CHAIN_IDs[key as keyof typeof CHAIN_IDs] === Number(spokeChainId)
+        );
+        if (spokeChainName) {
+          result[hubChainName][spokeChainName] = addresses;
+        }
+      }
+    }
+  }
+
+  return result;
+}
+
+// Helper function to convert OFT_EIDs to the expected format
+function convertOftEids(): { [key: string]: number } {
+  const result: { [key: string]: number } = {};
+
+  for (const [chainId, eid] of OFT_EIDs.entries()) {
+    const chainName = Object.keys(CHAIN_IDs).find(
+      (key) => CHAIN_IDs[key as keyof typeof CHAIN_IDs] === Number(chainId)
+    );
+    if (chainName) {
+      result[chainName] = eid;
+    }
+  }
+
+  return result;
+}
+
+// Helper function to convert CIRCLE_DOMAIN_IDs to the expected format
+function convertCircleDomainIds(): { [key: string]: number } {
+  const result: { [key: string]: number } = {};
+
+  for (const [chainId, domainId] of Object.entries(CIRCLE_DOMAIN_IDs)) {
+    const chainName = Object.keys(CHAIN_IDs).find(
+      (key) => CHAIN_IDs[key as keyof typeof CHAIN_IDs] === Number(chainId)
+    );
+    if (chainName) {
+      result[chainName] = domainId;
+    }
+  }
```

Comment:
This pattern is a bit repetitive now - is there maybe some way to refactor such that the core functionality is generic? Or are the upstream objects themselves too bespoke for that?